health/health_log.c

```diff
@@ -286,6 +302,8 @@ inline ssize_t health_alarm_log_read(RRDHOST *host, FILE *fp, const char *filena
             //    error("HEALTH [%s]: line %zu of file '%s' provides an alarm for host '%s' but this is named '%s'.", host->hostname, line, filename, pointers[1], host->hostname);
 
             ae->unique_id               = unique_id;
+            if (!is_valid_alarm_id(host, pointers[14], pointers[13], alarm_id))
+                alarm_id = rrdcalc_get_unique_id(host, pointers[14], pointers[13], NULL);
```

Comment:
`rrdcalc_get_unique_id()` will eventually execute `host->health_log.next_alarm_id++`. Is this a problem?

Reply:
I mean, is this going to create collisions?

Reply:
That should not be a problem by itself. There is code however that will force `host->health_log.next_alarm_id` host->health_log.next_alarm_id to the maximum found (L397) which would be bad in case one has been assigned. I will check that as well.

Reply:
What about concurrency - which threads can end up calling `rrdcalc_get_unique_id()` and do we need to guard the counter with a mutex as @underhood suggested?

Reply:
The function `health_alarm_log_read` calls `netdata_rwlock_rdlock(&host->health_log.alarm_log_rwlock);` before to load the file, at least for this function we should not have problems.